docs/features/runs/sweep-syntax.md

```diff
@@ -20,6 +22,12 @@ Passing in a flag without values is treated like a boolean
 grid run main.py --do_something
 ```
 
+Alternatively, your script can be written in such a way that it takes integers and acts on them as booleans. For instance, using a 1 or 0 will default to True or False respectively.
```

Comment:
We should prefer this way (`--opt False`) over the other (`--no-opt`).

> To perform grid search over boolean values, your script should be written such that it takes `True` or `False` values from the command line.
>
> e.g. `grid run foo.py --bar True` or `grid run foo.py --bar False`, NOT `grid run foo.py --bar` nor `grid run foo.py --no-bar`. This way, you can perform grid search with `grid run foo.py --bar '[True, False]'`
>
>If your script uses argparse, use the following template:
>```
>from distutils.utils import strtobool
>parser.add_argument('--opt1', dest='opt1', type=lambda x: bool(strtobool(x)))
>```
>
>Using `type=bool` is most likely not what you want: https://stackoverflow.com/questions/15008758/parsing->boolean-values-with-argparse

Reply:
@luca3rd I don't understand how the suggested change is substantially different from the provided example of using numeric values as your True/False. E.g

`grid run foo.py --bar 1 --barry 0`

In reality, it is up to the user to write their code to interpret the 1/0 or True/False correctly. I don't think this is a Grid preference thing but more a user preference thing. Can you elaborate on the benefit of the change?

Reply:
@luca3rd  `--opt` + `--no-opt` syntax isn't mentioned in this PR. Would this be sufficient?

"
Alternatively, your script can be written in such a way that it takes an arbitrary value and acts on them as booleans. For example, your script can be written in such a way that it treats 1/0, True, False, 'yes', 'no', etc as boolean values.
"

Reply:
I know it's not mentioned in this PR, but right above this chunk we have:
> ## Python flags
>
>### boolean
>Passing in a flag without values is treated like a boolean
>```text
>grid run main.py --do_something
>```

I feel like it's confusing to have both ways documented without explaining why you'd use one over the other. Like, in your example you don't mention the benefit of using our grid search syntax.

Reply:
Gotcha I understand now. I will make the change

Reply:
@luca3rd addressed.

Reply:
Looks good. Can you add an example for completeness? like just append `e.g. grid run foo.py --bar '[True, False]'` to the last line.

And it's up to you if you want to include the argparse snippet I linked above. But please add the grid search example and I'll approve.

Reply:
grid search example added @luca3rd